account_invoice_ubl/models/account_move.py

```diff
@@ -248,6 +280,8 @@ def generate_invoice_ubl_xml_etree(self, version="2.1"):
         xml_root = etree.Element("Invoice", nsmap=nsmap)
         self._ubl_add_header(xml_root, ns, version=version)
         self._ubl_add_order_reference(xml_root, ns, version=version)
+        self._ubl_add_buyer_reference(xml_root, ns, version=version)
+        self._ubl_add_invoice_order_reference(xml_root, ns, version=version)
```

Comment:
You're adding `OrderReference` here, but it's also added in `_ubl_add_order_reference` two lines above. You cannot add it twice, and it also should be added in the right order; so you'll have to remove this and modify the existing `_ubl_add_order_reference` instead.

Reply:
@thomaspaulb there is only one occurrence `_ubl_add_order_reference ` of this, the one I added and the other is in `sale_order_ubl` so we are only using `account_invoice_ubl` to add for invoice however I noticed its not required for validation and still green even after adding it. Think main issue was re-arrangement `_ubl_add_buyer_reference` coming before order

Reply:
Unfortunately this is still an issue, the guy who tested manages to create a condition where OrderReference is being added twice, and then he gets an error "OrderReference is not expected here" because [maxOccurrence is 1](https://github.com/sunflowerit/edi/blob/13.0-add-account_invoice_ubl_peppol/base_ubl/data/xsd-2.2/maindoc/UBL-Invoice-2.2.xsd#L441).

I think instead of adding a new function `_ubl_add_invoice_order_reference` we should just modify the existing `_ubl_add_order_reference` and choose which one we add. Both functions are in the same module anyway, in `account_invoice_ubl`, and even in the same file. (What does `sale_order_ubl` have to do with this?)

![image](https://user-images.githubusercontent.com/1466356/166139072-e5d21ff2-0927-4a74-82b1-a5a139d6569e.png)

Reply:
@thomaspaulb yeah you were right it creates a duplicate when we get an invoice from a sale order where the sale order name (i.e Origin in Invoice) is available. The invoice details already exist so there was no need to add them again. So I removed the piece code. Think its ok now.

Reply:
Cool, asking them to test again.